runtime/caml/compatibility.h

```diff
@@ -235,10 +235,6 @@
 /* **** meta.c */
 
 /* **** minor_gc.c */
```

Comment:
Is this huge change the appropriate to consider removing `compatibility.h` (or at least making its inclusion an a preprocessor error)?

Reply:
It may be the right time, but this should be a separate PR, that we'll need to check on OPAM and be ready to revert.

Reply:
ugh. The macro definitions such as `#define young_ptr ....` interferes with the field with the same name in domain state. If we are not modifying `compatibitlity.h` in this PR, then we should rename the domain state fields (`young_start`, `young_end`, `young_ptr`, `young_limit`, `local_roots`) to something else (by perhaps adding a prefix). Any suggestion on this?

Reply:
I reverted the change to `compatibility.h`. Now the PR doesn't modify compatibility.h.